crates/graph/src/graph.rs

```diff
@@ -93,7 +93,7 @@ pub enum LuminairError {
     StwoVerifierError(#[from] VerificationError),
 
     #[error("{0} LogUp values do not match.")]
-    InvalidLogUp(String),
+    InvalidLogUp(Box<str>),
```

Comment:
Removes 8 bytes from the `LuminairError`. Shrinking the size form 32 bytes to 24 bytes.

Error structures associated in the `Result` enum are passed in the stack on every function/method call, even if they are "unused". In extreme situations this can degrade runtime performance.

8 bytes probably won't make much of a difference in this project but it doesn't hurt to use `Box<str>`. The semantics remain the same.